File: frontend/lib/features/statistic/models/statistic_indicator.dart
class StatisticIndicator {
  final int id;
  final String title;
  final String? unit;
  final String? description;

  StatisticIndicator({
    required this.id,
    required this.title,
    this.unit,
    this.description,
  });

  factory StatisticIndicator.fromJson(Map<String, dynamic> json) {
    return StatisticIndicator(
      id: json['id'] as int,
      title: json['title'] as String,
      unit: json['unit'] as String?,
      description: json['description'] as String?,
    );
  }
}

File: backend/app/Models/StatisticIndicator.php
<?php

namespace App\Models;

use Illuminate\Database\Eloquent\Model;

class StatisticIndicator extends Model {

    protected $table = 'statistic_indicators';
    protected $primaryKey = 'id';

    protected $fillable = [
        'id',
        'subsubject_id',
        'title',
        'unit',
        'description'
    ];

    public function subsubject() {
        return $this->belongsTo(StatisticSubsubject::class);
    }

    public function data() {
        return $this->hasMany(StatisticData::class, 'indicator_id');
    }
}

